addons/website_forum/models/website.py

```diff
@@ -37,3 +39,18 @@ def _search_get_details(self, search_type, order, options):
         if search_type in ['forums', 'forum_posts_only', 'all']:
             result.append(self.env['forum.post']._search_get_detail(self, order, options))
         return result
+
+    def _update_forum_count(self):
+        """ Update count of forum linked to some websites. This has to be
+        done manually as website_id=False on forum model means a shared forum.
+        There is therefore no straightforward relationship to be used between
+        forum and website.
+
+        This method either runs on self (if not void), either on all existing
+        websites (to update globally counters, notably when a new forum is
+        created). """
+        websites = self if self else self.search([])
+        for website in websites:
+            website.forum_count = self.env['forum.forum'].sudo().search_count(
+                website.website_domain()
+            )
```

Comment:
```suggestion
        forums = self.env['forum.forum'].sudo().search([])
        for website in websites:
            website.forum_count = len(forums.filtered_domain(website.website_domain()))
```
Or we don't use website_domain.
Also, we shouldn't need sudo from caller code, right?

Reply:
Doesn't this avoid querying in a loop? (Also the test passes)

Reply:
Seems right, did not see the comment :) followup done at https://github.com/odoo/odoo/pull/116778